docs/internal-developers/block-client-apis/checkout/checkout-api.md

```diff
@@ -117,3 +117,92 @@ This hook is used to expose all the interfaces for the registered payment method
 _Why don't payment methods just implement this hook_?
 
 The contract is established through props fed to the payment method components via props. This allows us to avoid having to expose the hook publicly and experiment with how the props are retrieved and exposed in the future.
+
+## Examples
+
+### Passing a value from the client through to server side payment processing
+
+In this example, lets pass some data from the BACs payment method to the server. Registration of BACS looks like this:
+
+```js
+const bankTransferPaymentMethod = {
+	name: PAYMENT_METHOD_NAME,
+	label: <Label />,
+	content: <Content />,
+	edit: <Content />,
+	canMakePayment: () => true,
+	ariaLabel: label,
+	supports: {
+		features: settings?.supports ?? [],
+	},
+};
+```
+
+If we look a the `Content` component, we can see it defined as follows:
+
+```js
+const Content = () => {
+	return decodeEntities( settings.description || '' );
+};
+```
+
+Payment method components are passed, by default, everything from the `usePaymentMethodInterface` hook. So we can consume this in our component like so:
+
+```js
+const Content = ( props ) => {
+	const { eventRegistration, emitResponse } = props;
+	const { onPaymentProcessing } = eventRegistration;
+	useEffect( () => {
+		const unsubscribe = onPaymentProcessing( async () => {
+			// Here we can do any processing we need, and then emit a response.
+			// For example, we might validate a custom field, or perform an AJAX request, and then emit a response indicating it is valid or not.
+			const myGatewayCustomData = '12345';
+			const customDataIsValid = !! myGatewayCustomData.length;
+
+			if ( customDataIsValid ) {
+				return {
+					type: emitResponse.responseTypes.SUCCESS,
+					meta: {
+						paymentMethodData: {
+							myGatewayCustomData,
+						},
+					},
+				};
+			}
+
+			return {
+				type: emitResponse.responseTypes.ERROR,
+				message: 'There was an error',
+			};
+		} );
+		// Unsubscribes when this component is unmounted.
+		return () => {
+			unsubscribe();
+		};
+	}, [
+		emitResponse.responseTypes.ERROR,
+		emitResponse.responseTypes.SUCCESS,
+		onPaymentProcessing,
+	] );
+	return decodeEntities( settings.description || '' );
+};
+```
+
+Now when an order is placed, if we look at the API request payload, we can see the following JSON:
+
+```json
+{
+	"shipping_address": {},
+	"billing_address": {},
+	"customer_note": "",
+	"create_account": false,
+	"payment_method": "bacs",
+	"payment_data": [
+		{
+			"key": "myGatewayCustomData",
+			"value": "12345"
+		}
+	],
+	"extensions": {}
+}
+```
```

Comment:
Should we add how this is retrieved by the payment method on the server, or is that out of scope for this document?

Reply:
I think I'd like to keep this client focussed since that seems to be where most are struggling. Can revise in the future :)